cfme/tests/test_rest.py

```diff
@@ -69,6 +73,144 @@ def provision_data(
     return result
 
 
+@pytest.fixture(scope='module')
+def service_catalogs_fixture(request, rest_api):
+    name = fauxfactory.gen_alphanumeric()
+    scl = rest_api.collections.service_catalogs.action.add(
+        name=name,
+        description="description_{}".format(name),
+        service_templates=[]
+    )[0]
+    wait_for(
+        lambda: rest_api.collections.service_catalogs.find_by(name=name),
+        num_sec=180,
+        delay=10,
+    )
+
+    service_catalogs_data = [{
+        "name": "name_{}_{}".format(name, index),
+        "description": "description_{}_{}".format(name, index),
+        "service_templates": []
+    } for index in range(1, 5)]
+
+    scls = rest_api.collections.service_catalogs.action.add(*service_catalogs_data)
+
+    def _finished():
+        scl.reload()
+        scl.action.delete()
+        with error.expected("ActiveRecord::RecordNotFound"):
```

Comment:
The asserts should only be inside of the test run, not in teardown. If an assert fails inside the teardown, it makes the test end with ERROR state instead of FAILED state, meaning that something is wrong (like could not create a test vm, something that is required for the test could not be reached, ...).